src/main/java/net/snowflake/ingest/utils/SubColumnFinder.java

```diff
@@ -15,55 +14,94 @@
 import org.apache.parquet.schema.MessageType;
 import org.apache.parquet.schema.Type;
 
-/** Helper class to find all leaf sub-columns in an immutable schema given a dot path. */
+/** Helper class to find all leaf columns in an immutable schema given a fieldId. */
 public class SubColumnFinder {
-  static class SubtreeInterval {
+
+  /**
+   * Helper class to store the start and end index of the interval of leaf columns of a node in the
+   * list and the dot path of the node.
+   */
+  static class SubtreeInfo {
+    /* Start index of the leaf column in the list. */
     final int startTag;
+
+    /* End index of the leaf column in the list. */
     final int endTag;
 
-    SubtreeInterval(int startTag, int endTag) {
+    /* Dot path of the node. */
+    final String dotPath;
+
+    SubtreeInfo(int startTag, int endTag, String dotPath) {
       this.startTag = startTag;
       this.endTag = endTag;
+      this.dotPath = dotPath;
     }
   }
 
+  /* A list to store all leaf columns field id in preorder traversal. */
```

Comment:
Can the `list` also be List<Type.ID> now instead of List<String>?

Reply:
can do away with `new Type.ID(Integer.parseInt(subColumnId)) ` in ParquetRowBuffer if yes

Reply:
Yes but we need to do `toString()` every time when accessing `statsMap <String, Stats>`. I think either way works, do you have any preference?